### PR TITLE
Ensure edge.is_urban is set for serializing is_urban

### PR DESCRIPTION
**Pull request by danpaz**
# Issue

#2522 introduced a new is_urban flag to the osrm response, which uses the edge's `density` field to infer whether a road is considered to be urban or not. However, as it's an optional field, in the serializer we should be checking `edge.has_density` to ensure it has been populated before adding it to the response.

This also allows excluding is_urban flag from the response if you don't want it, by setting the exclude filter attribute for edge.density.
 

## Tasklist

 - [x] Add tests
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Generally use squash merge to rebase and clean comments before merging
 - [x] Update the [changelog](CHANGELOG.md)

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: test/urban.cc [hunk 1]
```diff
@@ -115,6 +115,34 @@ TEST(Urban2, test_density) {
...
+  route_tester tester;
+  std::string request =
+      R"({"locations":[{"lat":52.10160225589803,"lon":5.116925239562988},{"lat":52.09403591712907,"lon":5.113234519958496}],"costing":"auto","units":"miles","format":"osrm"})";
+  auto response = tester.test(request);
+
+  // get the osrm json
+  auto json_str = serializeDirections(response);
+  rapidjson::Document json;
+  json.Parse(json_str);
+  std::cout << "json :: " << json_str << std::endl;
```
Comment (mandeepsandhu) [inline test/urban.cc:129]: Leftover from debugging?

**Comment by danpaz** [comment]
After discussing with @kevinkreiser we decided to add a new filter attribute edge.is_urban rather than relying on the existing edge.density filter attribute. This allows us to make the information is_urban flag opt-in and won't have the unintended consequence of adding multiple properties to the tripleg with a single filter attribute.

cc @kdiluca @gknisely

File: test/urban.cc (hunk 1 above)
Comment (danpaz) [reply test/urban.cc:129]: Yep good catch

File: test/urban.cc [hunk 2]
```diff
@@ -115,6 +115,34 @@ TEST(Urban2, test_density) {
...
+  ASSERT_FALSE(json.HasParseError());
+
+  // loop over all routes all legs
+  for (const auto& route : json["routes"].GetArray()) {
+    for (const auto& leg : route["legs"].GetArray()) {
+      for (const auto& step : leg["steps"].GetArray()) {
+        for (const auto& intersection : step["intersections"].GetArray()) {
+          EXPECT_EQ(intersection.HasMember("is_urban"), false);
+        }
+        EXPECT_EQ(is_urban, true);
```
Comment (mandeepsandhu) [inline test/urban.cc:140]: What sets `is_urban` in this test? It seems to be left uninitialized.

Comment (danpaz) [reply test/urban.cc:138]: Leftover from rearranging the tests, thanks for catching 😄

File: test/urban.cc [hunk 3]
```diff
@@ -107,9 +106,35 @@ TEST(Urban2, test_density) {
...
           }
         }
-        EXPECT_EQ(is_urban, true);
+      }
+    }
+  }
+}
+
+TEST(Urban2, test_urban_excluded_by_default) {
+  bool is_urban;
```
Comment (mandeepsandhu) [inline test/urban.cc:118]: I guess we don't need this anymore?

**Pull request merged**